governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -0,0 +1,223 @@
+package com.netflix.governator.guice.test;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.ClassUtils;
+import org.mockito.Mockito;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Binding;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.binder.AnnotatedBindingBuilder;
+import com.google.inject.binder.LinkedBindingBuilder;
+import com.google.inject.name.Names;
+import com.google.inject.spi.DefaultElementVisitor;
+import com.google.inject.spi.Element;
+import com.google.inject.spi.Elements;
+import com.netflix.archaius.api.config.CompositeConfig;
+import com.netflix.archaius.api.config.SettableConfig;
+import com.netflix.archaius.config.DefaultSettableConfig;
+import com.netflix.archaius.guice.Raw;
+import com.netflix.archaius.test.TestCompositeConfig;
+import com.netflix.archaius.test.TestPropertyOverride;
+import com.netflix.archaius.test.TestPropertyOverrideAnnotationReader;
+import com.netflix.governator.InjectorBuilder;
+import com.netflix.governator.LifecycleInjector;
+import com.netflix.governator.providers.SingletonProvider;
+
+public class AnnotationBasedTestInjectorManager {
+
+    private final LifecycleInjector injector;
+    private final List<Object> mocksToReset = new ArrayList<>();
+    private final List<Module> modulesForTestClass = new ArrayList<>();
```

Comment:
The modulesForTestClass and overrideModules fields are only used in the constructor and the appropriate "inspect" method.  If the inspect methods returned the collection that they were building up, then these fields could be "localized" to the constructor.  You might then want to rename the "inspect<Blah>" methods back to "get<Blah>".